member-service/src/main/java/com/hedvig/memberservice/services/QualityAssuranceServiceImpl.kt

```diff
@@ -0,0 +1,52 @@
+package com.hedvig.memberservice.services
+
+import com.hedvig.external.zignSec.repository.ZignSecSignEntityRepository
+import com.hedvig.memberservice.query.SignedMemberRepository
+import com.hedvig.memberservice.services.exceptions.CountryNotFoundException
+import com.hedvig.memberservice.services.exceptions.SignedMemberNotFoundException
+import com.hedvig.memberservice.util.orNull
+import org.springframework.context.annotation.Profile
+import org.springframework.stereotype.Service
+
+@Service
+@Profile("staging", "development")
+class QualityAssuranceServiceImpl(
+    private val signedMemberRepository: SignedMemberRepository,
+    private val zignSecSignEntityRepository: ZignSecSignEntityRepository
+): QualityAssuranceService {
+    override fun unsignMember(country: String, ssn: String) =
+        when (country.toUpperCase()) {
+            "SWEDEN" -> unsignSwedishMember(ssn)
+            "NORWAY" -> unsignNorwegianMember(ssn)
+            else -> throw CountryNotFoundException(country)
+        }
+
+    private fun unsignSwedishMember(ssn: String): Boolean {
+        val signedMemberEntity = signedMemberRepository.findBySsn(ssn).orNull()
+            ?: throw SignedMemberNotFoundException(
+                ssn = ssn,
+                repositoryName = "SignedMemberRepository"
+            )
```

Comment:
I like your plan here but optional actually have a `orElseThrow` which you could use like this: 
```suggestion
        val signedMemberEntity = signedMemberRepository.findBySsn(ssn).orElseThrow {
            SignedMemberNotFoundException(
                ssn = ssn,
                repositoryName = "SignedMemberRepository"
            )
            }
```